### PR TITLE
Several things I find useful

### PR DESCRIPTION
**Pull request by hissssst**
* Monitoring without agent but with plain map instead
* Enforce keys in state struct
* Deprecation of `Poolex.get_state` in favor of `:sys.get_state`
* High priority for `Poolex` process

### CODE REVIEW

**Comment by general-CbIC** [comment]
Could you also describe your changes to CHANGELOG? 🙏🏻

**Comment by general-CbIC** [comment]
Closes #112 
Closes #115

File: lib/poolex.ex [hunk 1]
```diff
@@ -332,25 +319,20 @@ defmodule Poolex do
     {:noreply, state}
   end
 
-  @spec start_workers(non_neg_integer(), State.t()) :: [pid]
-  defp start_workers(0, _state) do
-    []
+  @spec start_workers(non_neg_integer(), State.t()) :: {[pid], State.t()}
+  defp start_workers(workers_count, state) when is_integer(workers_count) and workers_count >= 0 do
+    Enum.map_reduce(1..workers_count//1, state, fn _iterator, state ->
```
Comment (hissssst) [inline lib/poolex.ex:324]:
```suggestion
  defp start_workers(0, state) do
    {[], state}
  end

  defp start_workers(workers_count, state) when is_integer(workers_count) and workers_count >= 1 do
    Enum.map_reduce(1..workers_count, state, fn _iterator, state ->
```

**Comment by hissssst** [comment]
@general-CbIC rereview please

**Comment by general-CbIC** [comment]
Thanks for update!
I will merge it and fix warnings by myself.

**Pull request merged**